### PR TITLE
Refactor: Replace defaultProps with ES6 defaults

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves [GIVE-1109]

## Description

Replaces uses of `Component.defaultProps` with ES6 defaults.

## Testing Instructions

Build and spot-check for errors.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1109]: https://stellarwp.atlassian.net/browse/GIVE-1109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/DonorDashboards/resources/js/app/components/select-control/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import './style.scss';
 
 import {__} from '@wordpress/i18n';
 
-const SelectControl = ({label, value, isLoading, onChange, options, placeholder, width, isClearable}) => {
+const SelectControl = ({value, options, isLoading, label = null, onChange = null, placeholder = __('Select...', 'give'), width = null, isClearable = false}) => {
```
Comment (JoshuaHungDinh) [inline src/DonorDashboards/resources/js/app/components/select-control/index.js:14]: Do we need a default parameter for value here?

File: src/DonorDashboards/resources/js/app/components/subscription-manager/payment-method-control/square-control/index.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import './style.scss';
 const cardTokenizeResponse = {};
 let cardBrand = 'unknown';
 
-const SquareControl = ({label, value, forwardedRef, gateway}) => {
+const SquareControl = ({label = null, value, forwardedRef, gateway}) => {
```
Comment (JoshuaHungDinh) [inline src/DonorDashboards/resources/js/app/components/subscription-manager/payment-method-control/square-control/index.js:10]: Do we need a default parameter for value?

**Comment by kjohnson** [comment]
@JoshuaHungDinh I only migrated defaults that previously existed and did not want to introduce new defaults.

File: src/DonorDashboards/resources/js/app/components/subscription-manager/payment-method-control/square-control/index.js (hunk 2 above)
Comment (kjohnson) [reply src/DonorDashboards/resources/js/app/components/subscription-manager/payment-method-control/square-control/index.js:10]: See https://github.com/impress-org/givewp/pull/7490#issuecomment-2310556555

File: src/DonorDashboards/resources/js/app/components/select-control/index.js (hunk 1 above)
Comment (kjohnson) [reply src/DonorDashboards/resources/js/app/components/select-control/index.js:14]: See https://github.com/impress-org/givewp/pull/7490#issuecomment-2310556555

**Review by JoshuaHungDinh — APPROVED** [review]

**Comment by kjohnson** [comment]
Build for QA, see https://github.com/impress-org/givewp/actions/runs/10564819390.

**Review by rickalday — APPROVED** [review]
Passed manual and automated QA tests.

**Pull request merged**